crates/fuel-core/src/service.rs

```diff
@@ -3,6 +3,7 @@ use std::{
     sync::Arc,
 };
 
+use adapters::block_production_trigger::BlockProductionTrigger;
```

Comment:
I find it a bit confusing to reuse the "trigger" name for this synchronization primitive, as "trigger" already has a meaning for the PoA service.

If I understand this correctly, what we have here is just a type to signal that we're ready to start producing blocks. How about renaming it to something like `ReadySignal`? Or `StartBarrier`?

```suggestion
use adapters::block_production_trigger::ReadySignal;
```